### ISSUE TITLE
Hydra-unstable broken on master. Need lipqxx>=7

### ISSUE THREAD

**Issue by jonringer** [NixOS/hydra#978]
https://github.com/NixOS/nixpkgs/pull/126996 bumped `libqpxx` to 7.5.2 which breaks the build.

Why is this an issue for NixOS/hydra? 

The fixes don't look terrible to fix code side. And the PR removes one of the few remaining usages of python2.
I can pin another version of libpqxx until the fix lands here, but we should probably move to libpqxx anyway.

Error:

```
g++ -DHAVE_CONFIG_H -I. -I../..    -std=c++17 -I/nix/store/by2bh6wawngyxw4rha6l8yy5slq7pplc-boehm-gc-8.0.4-dev/include -I/nix/store/28c5xg650ig6yn2gc26hvra1d76fdjc4-nix-2.4pre20210601_5985b8b-dev/include/nix -Wall -I ../libhydra -Wno-deprecated-declarations -g -O2 -std=c++17 -include nix/config.h -c -o hydra_evaluator-hydra-evaluator.o `test -f 'hydra-evaluator.cc' || echo './'`hydra-evaluator.cc
In file included from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/util.hxx:31,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/result.hxx:25,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/binarystring.hxx:21,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/binarystring:4,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/pqxx:3,
                 from ../libhydra/db.hh:3,
                 from hydra-evaluator.cc:1:
../../version:1:1: error: expected unqualified-id before numeric constant
    1 | 0.1
      | ^~~
In file included from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/util.hxx:37,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/result.hxx:25,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/binarystring.hxx:21,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/binarystring:4,
                 from /nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/pqxx:3,
                 from ../libhydra/db.hh:3,
                 from hydra-evaluator.cc:1:
/nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/util.hxx: In function 'void pqxx::check_version()':
/nix/store/0p9hqklkykhrig45y8r5iw8ysw4q83vk-libpqxx-7.5.2/include/pqxx/util.hxx:183:42: error: 'check_pqxx_version_PQXX_VERSION_MAJOR_PQXX_VERSION_MINOR' is not a member of 'pqxx::internal'
  183 |   static auto const version_ok{internal::PQXX_VERSION_CHECK()};
      |                                          ^~~~~~~~~~~~~~~~~~
hydra-evaluator.cc: In member function 'bool Evaluator::shouldEvaluate(Evaluator::Jobset&)':
hydra-evaluator.cc:236:43: error: 'using work = class pqxx::transaction<>' {aka 'class pqxx::transaction<>'} has no member named 'parameterized'
  236 |                 auto evaluation_res = txn.parameterized
      |                                           ^~~~~~~~~~~~~
hydra-evaluator.cc:250:62: error: expected primary-expression before 'int'
  250 |                 auto evaluation_id = evaluation_res[0][0].as<int>();
      |                                                              ^~~
hydra-evaluator.cc:252:49: error: 'using work = class pqxx::transaction<>' {aka 'class pqxx::transaction<>'} has no member named 'parameterized'
  252 |                 auto unfinished_build_res = txn.parameterized
      |                                                 ^~~~~~~~~~~~~
      ```

**Comment by Taneb** [comment]
There's a few things going on here:

1. we use some deprecated functionality (the `txt::parameterized` thing). This is easy to replace
2. https://github.com/jtv/libpqxx/issues/459 (the `'check_pqxx_version_PQXX_VERSION_MAJOR_PQXX_VERSION_MINOR' is not a member of 'pqxx::internal'` error). This is fixed in pqxx HEAD but not in the released version
3. Some weirdness due to `<version>` being a standard header in C++20 onwards which libpqxx attempts to include if it can. And it _can_ include it, because the root of the Hydra package is on the include path, but it gets the wrong kind of file entirely.

I've got a patch fixing 1 and 3 (by moving `version` to `version.txt`), and I've checked it builds against libpqxx HEAD. I'll make a PR this eveing.
